crates/solver/src/settlement_rater.rs

```diff
@@ -339,7 +390,10 @@ impl ScoreCalculator {
             cost_fail,
             self.score_cap.clone(),
         )?;
-        let score = big_rational_to_u256(&optimal_score).context("Invalid score.")?;
+        if optimal_score > *objective_value {
+            return Err(ScoringError::ScoreHigherThanObjective(optimal_score));
+        }
```

Comment:
Here we compute the optimal score on behalf of the solver using the objective value and success probability.
Since we already enforced that the objective value is positive and that success is in `[0,1]` this check would indicate an error in our optimal score computation logic, right?
Meaning if a solver were to get this error they would have no way of "fixing" the issue.

Reply:
You are right. But it's hard to draw a line which errors are useful to solvers and which aren't. I don't think we should notify them only for errors that they can action upon. OTOH there is no much point in notifying them about our web3 connection issues.
So I just decided to notify them whenever we can give them reasonable error that they can understand.

Reply:
If this actually indicates a logic bug in our score computation logic should we maybe add an `error` log here. Otherwise we'd have to periodically check our warning logs or hope that solvers report that error back to us to make us aware.